lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2896,6 +2896,60 @@ class DecomposeAtenMseLossOp : public OpRewritePattern<AtenMseLossOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenRandintLowOp : public OpRewritePattern<AtenRandintLowOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenRandintLowOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    Type resultType = op.getType();
+    BaseTensorType resultTensorType = resultType.cast<BaseTensorType>();
+
+    int64_t cstLow, cstHigh;
+    if (!matchPattern(op.low(), m_TorchConstantInt(&cstLow)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: low must be a constant integer");
+    if (!matchPattern(op.high(), m_TorchConstantInt(&cstHigh)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: high must be a constant integer");
+
+    Value none = rewriter.create<ConstantNoneOp>(loc);
+    Value cstFalse = rewriter.create<ConstantBoolOp>(loc, false);
+    Value zero =
+        rewriter.create<ConstantFloatOp>(loc, rewriter.getF64FloatAttr(0.0));
+    Value low = rewriter.create<Torch::ConstantFloatOp>(
+        loc, rewriter.getF64FloatAttr((double)cstLow));
+    Value high = rewriter.create<Torch::ConstantFloatOp>(
+        loc, rewriter.getF64FloatAttr((double)cstHigh));
+    Value emptyTensor = rewriter.create<AtenFullOp>(
+        loc, resultType, op.size(), zero, op.dtype(), op.layout(), op.device(),
+        op.pin_memory());
+
+    BaseTensorType floatResultType =
+        resultTensorType
+            .getWithSizesAndDtype(resultTensorType.getSizes(),
+                                  rewriter.getF64Type())
+            .cast<BaseTensorType>();
+    emptyTensor = rewriter.create<AtenToDtypeOp>(
+        loc, floatResultType, emptyTensor,
+        getDtypeIntValueForType(rewriter, loc, rewriter.getF64Type()), cstFalse,
+        cstFalse, none);
```

Comment:
I don't think this `AtenToDtypeOp` is needed. You should be able to get the right type for the `emptyTensor` by passing the right result type and `dtype` arguments to `AtenFullOp`